### DISCUSSION TITLE
undeterministic time of search when num_search_workers set to 8

### DISCUSSION THREAD

**Discussion by DrXuQian** [google/or-tools#3852] [CP-SAT questions] [answered]
**What version of OR-Tools and what language are you using?**
Version: main
Language: Python

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
CP-SAT
**What operating system (Linux, Windows, ...) and version?**
Ubuntu
**What did you do?**
For the same problem, I get undeterministic time of search for multiple runs. Sometimes the search finishes in a few seconds and sometime the search can't finish in 10minutes timeout.

**What did you expect to see**
I wonder if this is supposed to be like this when use multi-process or not. 
I am also curious what I can do in such condition to accelerate the search. 

Model proto:
[cp_sat_optimize_prefetch.txt](https://github.com/google/or-tools/files/11980344/cp_sat_optimize_prefetch.txt)

Thanks!

**Comment by gregy4** [comment, marked as answer]
Undeterministic execution time has probably two reasons. One is usage of lns with randomness, it was possible to deny usage of lns when num_search_workers = 6. Second reason is a variation in order of solutions between search workers, sometimes two or more search workers found a solution in similar time and between runs it can vary. Better solutions can differ from this point.

For me minimalization of variations between runs is part of optimization.
